app/src/main/java/org/adaway/ui/prefs/BasePreferenceFragment.java

```diff
@@ -0,0 +1,66 @@
+package org.adaway.ui.prefs;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.preference.EditTextPreference;
+import androidx.preference.ListPreference;
+import androidx.preference.Preference;
+import androidx.preference.PreferenceFragmentCompat;
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+import com.google.android.material.textfield.TextInputEditText;
+import org.adaway.R;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BasePreferenceFragment extends PreferenceFragmentCompat {
+    @Override
+    public void onCreatePreferences(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState, @Nullable @org.jetbrains.annotations.Nullable String rootKey) {}
+
+    @Override
+    public void onDisplayPreferenceDialog(@NonNull Preference preference) {
+        if (preference instanceof ListPreference) {
+            showListPreference((ListPreference) preference);
+        } else if (preference instanceof EditTextPreference) {
+            showEditTextPreference((EditTextPreference) preference);
+        } else {
+            super.onDisplayPreferenceDialog(preference);
+        }
+    }
```

Comment:
Is this required to use MD components by default?

Reply:
Yes, that's needed for the preferences to look material 3 designed.
Preferences are not yet officially available at Material Design 3, so we have to override the default Material Design 2 preference dialogs and show our own custom, MaterialAlertDialogBuilder instances instead, which use MD3.